Branching from parent without changing name raises RaceCondition

If a parent and a child experiment with the same name already exists, branching from the parent without changing the name will lead to a RaceCondition error.

```python
            try:
                _register_experiment(branched_experiment)
                log.debug("Branched experiment successfully registered in DB.")
            except DuplicateKeyError as e:
                log.debug(
                    "Experiment registration failed. This is likely due to a race condition "
                    "during branching. Now rolling back and re-attempting building "
                    "the branched experiment."
                )
>               raise RaceCondition("There was a race condition during branching.") from e
E               orion.core.utils.exceptions.RaceCondition: There was a race condition during branching.
```

This should be detected as a conflict that cannot be resolved because the name of the experiment must be changed. The error message should be explicit about this. It is not a race condition.

--- a/src/orion/core/io/experiment_builder.py
+++ b/src/orion/core/io/experiment_builder.py
@@ -612,7 +612,12 @@ def _attempt_branching(conflicts, experiment, version, branching):
             "during branching. Now rolling back and re-attempting building "
             "the branched experiment."
         )
-        raise RaceCondition("There was a race condition during branching.") from e
+        raise RaceCondition(
+            "There was a race condition during branching. This error can "
+            "also occur if you try branching from a specific version that already "
+            "has a child experiment with the same name. Change the name of the new "
+            "experiment and use `branch-from` to specify the parent experiment."
+        ) from e
 
     return branched_experiment
 

